### PR TITLE
Testing OSX in Travis CI

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by certik** [comment]
Nice!! Yes, exactly. We need to test both clang and gcc on mac, as we found
out with cmake, some packages build with one, but not the other. Eventually
we'll build all optional dependencies as well.

**Comment by certik** [comment]
It seems the Python tests with clang + osx at the end here: https://travis-ci.org/sympy/symengine/jobs/67358268 are not being run.

**Comment by isuruf** [comment]
That's because we didn't enable it. At the moment, when installing via `make install` instead of `python setup.py install` we don't copy the python tests. So, tests are not run.

**Comment by isuruf** [comment]
gcc test is failing. We have to fix. #367

**Comment by isuruf** [comment]
Ready for review now

**Comment by isuruf** [comment]
I upgraded arb to 2.6.0, because of a warning of comparing long and unsigned long in an arb header. It's weird that no warning was given earlier

File: bin/install_travis.sh [hunk 1]
```diff
@@ -1,37 +1,82 @@
...
 
-sudo apt-get update
-sudo apt-get install cmake libgmp-dev
+if [[ "${CC}" == "" ]]; then
+    export CC=gcc
+    export CXX=g++
+fi
+if [[ "${TRAVIS_OS_NAME}" == "osx" ]] && [[ "${CC}" == "gcc" ]]; then
+    export CC=gcc-4.8
+    export CXX=g++-4.8
```
Comment (certik) [inline bin/install_travis.sh]: Cannot you just say `gcc` and `g++`? Why do you need to specify a particular version? If Travis updates, we'll have to change this.

**Comment by certik** [comment]
This looks good. +1 Make sure Shippable passes before merging.

File: bin/install_travis.sh (hunk 1 above)
Comment (isuruf) [inline bin/install_travis.sh]: `gcc` is just a wrapper around `clang` whereas `gcc-4.8` is a true `gcc`.

https://github.com/travis-ci/travis-ci/issues/2423

Comment (certik) [inline bin/install_travis.sh]: Ah, that's right. I forgot about that.

**Comment by isuruf** [comment]
Thanks for the review. Tests are passing on Shippable and Travis-CI

**Pull request merged**